### PR TITLE
Segmentation Tutorial

### PR DESCRIPTION
**Pull request by paulguerrie**
Added a `tutorial.ipynb` to `segment/` to demonstrate training, validation, and prediction for YOLOv5 Segmentation.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
New Jupyter notebook tutorial for YOLOv5 segmentation capabilities.

### 📊 Key Changes
- Addition of a comprehensive tutorial notebook (`tutorial.ipynb`) for YOLOv5 segmentation.
- Tutorial covers setup, prediction examples, validation on COCO dataset, training, and visualization. 

### 🎯 Purpose & Impact
- **Ease of Use**: Allows users to follow step-by-step instructions to use YOLOv5 for segmentation tasks.
- **Educational Value**: Helps new users understand YOLOv5's segmentation process, from installation to training and inference.
- **Performance Verification**: Users can validate model accuracy on standard segmentation datasets, ensuring reliability and robustness.
- **Customizability**: The tutorial demonstrates training on custom datasets with Roboflow, fostering adaptation to diverse usage scenarios.
- **Reproducibility**: Integrated logging and third-party platforms (Comet, ClearML) support experiment tracking and model training reproducibility.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@paulguerrie awesome, looks much better :)

**Comment by glenn-jocher** [comment]
unnamed

**Comment by Sarveshsh8** [comment]
How to run a inference just like object detection?
model = torch.hub.load()
is there any method to run a inference like this for segmentation?

**Comment by glenn-jocher** [comment]
@Sarveshsh8 yes, for segmentation you can run inference by loading a model and passing inputs through it. Here's an example using the YOLOv5 model:

```python
import torch
from PIL import Image

# Load the model
model = torch.hub.load('ultralytics/yolov5', 'custom', path_or_model='best.pt')

# Load an image
img = Image.open('path/to/image.jpg')

# Perform inference
results = model(img)

# Display results
results.show()
```

For more detailed usage, check out the official [Ultralytics YOLOv5 documentation](https://docs.ultralytics.com/yolov5/).
